src/main/java/ca/gc/aafc/dinauser/api/service/DinaUserService.java

```diff
@@ -316,4 +321,71 @@ public void deleteUser(final String id) {
     res.remove();
   }
 
+  @Override
+  public DinaUserDto create(DinaUserDto entity) {
+    DinaUserDto user = this.createUser(entity);
+    // Small flaw in dina repo, need to set id on incoming entity
+    entity.setInternalId(user.getInternalId());
+    return user;
+  }
+
+  @Override
+  public DinaUserDto update(DinaUserDto entity) {
+    return this.updateUser(entity);
+  }
+
+  @Override
+  public void delete(DinaUserDto entity) {
+    this.deleteUser(entity.getInternalId());
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> T findOne(Object naturalId, Class<T> entityClass) {
+    validateFindClass(entityClass);
+    return (T) this.getUser(naturalId.toString());
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> T findOneReferenceByNaturalId(Class<T> entityClass, Object naturalId) {
+    validateFindClass(entityClass);
+    return (T) this.getUser(naturalId.toString());
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> List<T> findAll(
+    @NonNull Class<T> entityClass,
+    @NonNull BiFunction<CriteriaBuilder, Root<T>, Predicate[]> where,
+    BiFunction<CriteriaBuilder, Root<T>, List<Order>> orderBy,
+    int startIndex,
+    int maxResult
+  ) {
+    validateFindClass(entityClass);
+    return (List<T>) this.getUsers();
+  }
+
+  @Override
+  public <T> Long getResourceCount(
+    @NonNull Class<T> entityClass,
+    @NonNull BiFunction<CriteriaBuilder, Root<T>, Predicate[]> predicateSupplier
+  ) {
+    validateFindClass(entityClass);
+    return (long) this.getUsers().size();
+  }
+
+  @Override
+  public boolean exists(Class<?> entityClass, Object naturalId) {
+    validateFindClass(entityClass);
+    return this.getUsers().stream().map(DinaUserDto::getInternalId).anyMatch(
```

Comment:
Since this is just using the internalId, why not .getUser and check if it's not null? That seems more efficient than going through the entire user list.

Reply:
Yes that is much better.